### ISSUE TITLE
duckdb-rs 0.10.0 compilation failed on windows 11

### ISSUE THREAD

**Issue by ou-bing** [duckdb/duckdb-rs#268]
![image](https://github.com/duckdb/duckdb-rs/assets/32380374/34693827-698b-4051-9382-ddbc8415f718)

OS: Windows 11 23H2
RUST Toolchains: rustc 1.76.0 (07dca489a 2024-02-04)

err msg:
```
  error occurred: Command "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.38.33130\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-I" "duckdb" "-I" "duckdb/third_party/concurrentqueue" "-I" "duckdb/third_party/utf8proc/include" "-I" "duckdb/third_party/fmt/include" "-I" "duckdb/third_party/fast_float" "-I" "duckdb/third_party/mbedtls/include" "-I" "duckdb/third_party/libpg_query" "-I" "duckdb/third_party/re2" "-I" "duckdb/third_party/hyperloglog" "-I" "duckdb/src/include" "-I" "duckdb/third_party/mbedtls" "-I" "duckdb/third_party/libpg_query/include" "-I" "duckdb/third_party/httplib" "-I" "duckdb/third_party/tdigest" "-I" "duckdb/third_party/jaro_winkler" "-I" "duckdb/third_party/miniz" "-I" "duckdb/third_party/utf8proc" "-I" "duckdb/third_party/fsst" "-I" "duckdb/third_party/fastpforlib" "-I" "duckdb/third_party/mbedtls/library" "-I" "duckdb/third_party/skiplist" "-I" "duckdb/third_party/pcg" "-I" "duckdb/third_party/jaro_winkler/details" "/bigobj" "-DDUCKDB_EXTENSION_AUTOINSTALL_DEFAULT=1" "-DDUCKDB_EXTENSION_AUTOLOAD_DEFAULT=1" "-DDUCKDB_BUILD_LIBRARY" "-FoD:\\src\\demo\\demo-duckdb\\target\\debug\\build\\libduckdb-sys-5d16083e927cb282\\out\\239a61d5cb207136-ub_src_main.o" "-c" "duckdb/ub_src_main.cpp" with args "cl.exe" did not execute successfully (status code exit code: 2).
```

**Comment by photino** [comment]
I also failed to build `libduckdb-sys v0.10.2` on Windows.

**Comment by Mause** [comment]
Given there's no actual error in the screenshot or your message, there's not a lot we can do to help

**Comment by eaglebetter** [comment]
error occurred: Command "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-I" "duckdb" "-I" "duckdb/src/include" "-I" "duckdb/third_party/jaro_winkler/details" "-I" "duckdb/third_party/hyperloglog" "-I" "duckdb/third_party/libpg_query" "-I" "duckdb/third_party/httplib" "-I" "duckdb/third_party/fast_float" "-I" "duckdb/third_party/mbedtls/include" "-I" "duckdb/third_party/utf8proc/include" "-I" "duckdb/third_party/fmt/include" "-I" "duckdb/third_party/re2" "-I" "duckdb/third_party/mbedtls" "-I" "duckdb/third_party/jaro_winkler" "-I" "duckdb/third_party/libpg_query/include" "-I" "duckdb/third_party/fsst" "-I" "duckdb/third_party/tdigest" "-I" "duckdb/third_party/fastpforlib" "-I" "duckdb/third_party/utf8proc" "-I" "duckdb/third_party/mbedtls/library" "-I" "duckdb/third_party/concurrentqueue" "-I" "duckdb/third_party/miniz" "-I" "duckdb/third_party/pcg" "/bigobj" "-DDUCKDB_EXTENSION_AUTOINSTALL_DEFAULT=1" "-DDUCKDB_EXTENSION_AUTOLOAD_DEFAULT=1" "-DDUCKDB_BUILD_LIBRARY" "-FoD:\\docker\\rust\\cpp_to_rust\\rust_project\\target\\debug\\build\\libduckdb-sys-35e7a27cfc7a2274\\out\\239a61d5cb207136-ub_src_main.o" 
"-c" "duckdb/ub_src_main.cpp" with args "cl.exe" did not execute successfully (status code exit code: 2).

I also encountered the compilation error mentioned above，May I ask how to obtain more compilation error information？

**Comment by photino** [comment]
I have spent more than 3 hours, but never succeed to compile it ☹️

**Comment by Mause** [comment]
Until someone provides a full compilation log, there's not a lot we can do to help

**Comment by wyhaya** [comment]
I'm experiencing the same error on a brand new Windows 10, but have no problem with GitHub actions ci, maybe some dependencies are missing?

**Comment by zhajingzhaxi** [comment]
@wyhaya I have the same problem as you on Windows 10, have you solved it？

**Comment by wyhaya** [comment]
> @wyhaya I have the same problem as you on Windows 10, have you solved it？

No, I still think something is missing or the environment is not compatible, because it works fine in GitHub Action. I don't know much about C++, but if possible, you could try compiling [duckdb/duckdb](https://github.com/duckdb/duckdb) directly to see if you can find anything.

**Comment by zhajingzhaxi** [comment]
> > @wyhaya I have the same problem as you on Windows 10, have you solved it？
> 
> No, I still think something is missing or the environment is not compatible, because it works fine in GitHub Action. I don't know much about C++, but if possible, you could try compiling [duckdb/duckdb](https://github.com/duckdb/duckdb) directly to see if you can find anything.

thanks

**Comment by photino** [comment]
Is it fixed in `duckdb-rs 1.0`?

**Comment by tansen87** [comment]
Using GNU on Windows can solve compilation error problems.
You can follow [zhihu](https://zhuanlan.zhihu.com/p/613466378).

**Comment by wuuer** [comment]
removing "bundled" features solves the problem

**Comment by dongbin86** [comment]
@tansen87 After switching to GNU, I indeed solved this problem.
